src/module_2/eda_groceries.md

```diff
@@ -0,0 +1,506 @@
+
+
+```python
+variant_price = (
+    df_orders.explode("variant_id")
+    .reset_index(drop=True)
+    .merge(df_inventory.loc[:, ["variant_id", "price"]], on="variant_id", how="left")
+)
+# variant_price -> [id, user_id, created_at, variant_id, price]
+order_revenue = variant_price.groupby("id")["price"].sum().reset_index()
+print(order_revenue.describe())
+order_revenue["price"].hist(bins=100, log=True)
+plt.xlabel("Total Revenue per Order")
+plt.ylabel("Frequency")
+```
+
+                     id        price
+    count  5.457000e+03  5457.000000
+    mean   2.161881e+13    30.316320
+    std    4.028679e+11    21.473383
+    min    1.285856e+13     0.000000
+    25%    2.133401e+13    14.140000
+    50%    2.167062e+13    26.870000
+    75%    2.192303e+13    42.210000
+    max    2.223385e+13   311.790000
+
+
+
+
+
+    Text(0, 0.5, 'Frequency')
+
+
+
+
+    
+![png](eda_groceries_files/eda_groceries_12_2.png)
+    
+
+
+
+```python
+variant_price.groupby("created_at")["price"].sum().plot(kind="bar")
+plt.title("Revenue by month")
+plt.xlabel("Month")
+plt.ylabel("Total revenue")
+```
+
+
+
+
+    Text(0, 0.5, 'Total revenue')
+
+
+
+
+    
+![png](eda_groceries_files/eda_groceries_13_1.png)
+    
+
+
+
+```python
+# Products prices distribution
+df_inventory["price"].hist(bins=100, log=True)
+df_inventory.describe()
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>variant_id</th>
+      <th>price</th>
+      <th>compare_at_price</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>count</th>
+      <td>1.733000e+03</td>
+      <td>1733.000000</td>
+      <td>1733.000000</td>
+    </tr>
+    <tr>
+      <th>mean</th>
+      <td>3.694880e+13</td>
+      <td>6.307351</td>
+      <td>7.028881</td>
+    </tr>
+    <tr>
+      <th>std</th>
+      <td>2.725674e+12</td>
+      <td>7.107218</td>
+      <td>7.660542</td>
+    </tr>
+    <tr>
+      <th>min</th>
+      <td>3.361529e+13</td>
+      <td>0.000000</td>
+      <td>0.000000</td>
+    </tr>
+    <tr>
+      <th>25%</th>
+      <td>3.427657e+13</td>
+      <td>2.490000</td>
+      <td>2.850000</td>
+    </tr>
+    <tr>
+      <th>50%</th>
+      <td>3.927260e+13</td>
+      <td>3.990000</td>
+      <td>4.490000</td>
+    </tr>
+    <tr>
+      <th>75%</th>
+      <td>3.948318e+13</td>
+      <td>7.490000</td>
+      <td>8.210000</td>
+    </tr>
+    <tr>
+      <th>max</th>
+      <td>4.016793e+13</td>
+      <td>59.990000</td>
+      <td>60.000000</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+    
+![png](eda_groceries_files/eda_groceries_14_1.png)
```

Comment:
Perfect example: Im missing axis labels and title; the graph is hard to interpret unless I read the code

Reply:
There's actually no code that produces this plot, is there?